### PR TITLE
Modin DF support

### PR DESCRIPTION
**Pull request by ShvetsKS**
Added support of optimized Pandas DataFrame - Modin: https://github.com/modin-project/modin
it helps to improve data pre-processing time

### CODE REVIEW

**Review by trivialfis — COMMENTED** [review]
Please add it to CI.

**Comment by ShvetsKS** [comment]
> Please add it to CI.

@trivialfis  I hope changes in CI specific part are correct.

**Comment by trivialfis** [comment]
@hcho3 Please help taking a look into the CI changes.  I thing some installations are redundant but can't be sure.

**Comment by hcho3** [comment]
Fixed.

File: tests/python/test_with_modin.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+import unittest
+import pytest
+
+try:
+    import modin.pandas as md
+except ImportError:
+    pass
+
+
+pytestmark = pytest.mark.skipif(**tm.no_pandas())
```
Comment (hcho3) [inline tests/python/test_with_modin.py:14]: Reminder: Add a skip marker for missing modin.

**Comment by hcho3** [comment]
Modin is interfering with Dask somehow.

**Comment by hcho3** [comment]
This reminds me of what happened with datatable...

**Comment by ShvetsKS** [comment]
> Modin is interfering with Dask somehow.

@hcho3 , @trivialfis seems all tests are passed.
To fix interfering with Dask the user should follow instruction provided here: https://github.com/modin-project/modin
just set env variable before import:
`os.environ["MODIN_ENGINE"] = "dask"  # Modin will use Dask`

**Review by trivialfis — APPROVED** [review]

**Review by SmirnovEgorRu — APPROVED** [review]

**Comment by ShvetsKS** [comment]
@trivialfis , @SmirnovEgorRu sorry for committing changes after approves
it's small fix to reduce extra conversion time from pandas to expected numpy np.float32

File: python-package/xgboost/training.py [hunk 2]
```diff
@@ -369,7 +369,7 @@ def aggcv(rlist):
 def cv(params, dtrain, num_boost_round=10, nfold=3, stratified=False, folds=None,
        metrics=(), obj=None, feval=None, maximize=False, early_stopping_rounds=None,
        fpreproc=None, as_pandas=True, verbose_eval=None, show_stdv=True,
-       seed=0, callbacks=None, shuffle=True):
+       seed=0, callbacks=None, shuffle=True, as_modin=False):
```
Comment (trivialfis) [inline python-package/xgboost/training.py:372]: Can we remove changes to cv for now?  I don't like this function and seeking for external implementation, or at least a better re-implementation.

**Review by trivialfis — CHANGES_REQUESTED** [review]

File: python-package/xgboost/training.py (hunk 2 above)
Comment (ShvetsKS) [reply python-package/xgboost/training.py:372]: yes. But seems I should also delete related test case for Modin, is it ok?

Comment (trivialfis) [reply python-package/xgboost/training.py:372]: That's fine.

Comment (ShvetsKS) [reply python-package/xgboost/training.py]: done :)

**Review by trivialfis — APPROVED** [review]

**Pull request merged**